sphinx_needs/directives/needflow.py

```diff
@@ -303,11 +303,12 @@ def process_needflow(app: Sphinx, doctree: nodes.document, fromdocname: str, fou
         all_needs = env.needs_all_needs
 
         option_link_types = [link.upper() for link in current_needflow["link_types"]]
+        allowed_link_types = [link["option"].upper() for link in link_types]
```

Comment:
Is this correct?
`link_types` are the currently set link types from the option value.
So you check if the given link types are in the given link types, or not?

I would go with ` app.config.needs_extra_links` to check if a given link type got also registered in Sphinx-Needs itself.

Reply:
To be honest, I just copied the list comprehension from the if statement at line 307 and assigned it to a variable which is then referenced in the join on line 311 because I assumed that the message should probably reflect why the if condition was entered in the first place. 
But I have tested it and the code change did fix the crash for me and throws (what I feel like is) a reasonable looking log message like seen in the attached ticket.